core/spot_termination.go

```diff
@@ -101,3 +116,57 @@ func (s *SpotTermination) getAsgName(instanceID *string) (string, error) {
 
 	return asgName, err
 }
+
+// ExecuteAction execute the proper termination action (terminate|detach) based on the value of
+// terminationNotificationAction and the presence of a LifecycleHook on ASG.
+func (s *SpotTermination) ExecuteAction(instanceID *string, terminationNotificationAction string) error {
+	if s.asSvc == nil {
+		return errors.New("AutoScaling service not defined. Please use NewSpotTermination()")
+	}
+
+	asgName, err := s.getAsgName(instanceID)
+
+	if err != nil {
+		log.Printf("Failed get ASG name for %s with err: %s\n", *instanceID, err.Error())
+		return err
+	}
+
+	switch terminationNotificationAction {
+	case "detach":
+		s.detachInstance(instanceID, asgName)
+	case "terminate":
+		s.terminateInstance(instanceID, asgName)
+	default:
+		if s.asgHasHook(&asgName) {
+			s.terminateInstance(instanceID, asgName)
+		} else {
+			s.detachInstance(instanceID, asgName)
+		}
+	}
+
+	return nil
+}
+
+func (s *SpotTermination) asgHasHook(autoScalingGroupName *string) bool {
```

Comment:
I would rename this to `asgHasTerminationLifecycleHook` or if we keep the name as it is, we can make it take the hook type as an additional parameter.

Reply:
Ok. As the only LF that make sense, in that contest, is the Termination one i will rename it as your suggestion.